lib/pure/bitops.nim

```diff
@@ -32,6 +32,22 @@ const useICC_builtins = defined(icc) and useBuiltins
 const useVCC_builtins = defined(vcc) and useBuiltins
 const arch64 = sizeof(int) == 8
 
+template toUint32[T](x: T): uint32 =
+  when sizeof(x) == 1: cast[uint8](x).uint32
+  elif sizeof(x) == 2: cast[uint16](x).uint32
+  else:                cast[uint32](x)
```

Comment:
I think this should be ``elif sizeof(x) == 4: cast[uint32](x) else: {.error: "invalid size".}``  and likewise for the other parts where this pattern is used.

Reply:
not relevant as of 1.3.5 278b4582f5c7cc3471bccc4ba587030edf7a50e2

Reply:
oups, didn't realize PR was closed but not merged